app/code/Magento/MediaGallery/etc/db_schema.xml

```diff
@@ -8,7 +8,7 @@
 <schema xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Setup/Declaration/Schema/etc/schema.xsd">
     <table name="media_gallery_asset" resource="default" engine="innodb" comment="Media Gallery Asset">
         <column xsi:type="int" name="id" unsigned="true" nullable="false" identity="true" comment="Entity ID"/>
-        <column xsi:type="varchar" name="path" length="255" nullable="true" comment="Path"/>
+        <column xsi:type="text" name="path" nullable="true" comment="Path"/>
```

Comment:
I think it's better to introduce validation on the folder name length instead of changing the column type since the file system has limitations too. Information for the reference: https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits

Reply:
but this is not folder name this is full path like /filder1/folde2/looooooooooooooooooooooooongfolder